### PR TITLE
Refactor applet config validation

### PR DESCRIPTION
**Pull request by mleotta**
Applets reuse the same pattern repeatedly of checking configuration for input or output files and directories.  Depending on whether the input or output is required or optional the logic is slightly different, but generally we want to see if the parameter is set and if it points to a valid file or directory on disk.  For outputs, we may even want to create missing directories and test that we have permission to write to output files.

This PR refactors current applets to move this logic into a set of reusable utility functions that any applet can reuse.

### CODE REVIEW

File: vital/applets/config_validation.h [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+#define KWIVER_TOOLS_CONFIG_VALIDATION_H
+
+#include <vital/applets/kwiver_tools_applet_export.h>
+
+#include <vital/config/config_block.h>
+#include <vital/logger/logger.h>
+
+#include <string>
+
+namespace kwiver {
```
Comment (mleotta) [inline vital/applets/config_validation.h:41]: @linus-sherrill I was following the established naming here, but this namespace and library name seems wrong given that this is in `vital\applets`.  Shouldn't the namespace be `kwiver::vital` or `kwiver::vital::applets`?

**Review by dstoup — APPROVED** [review]
I had the same question as Linus. Otherwise, looks good!

**Comment by mleotta** [comment]
I'd like to merge this as is (consistent with existing namespaces) and then maybe we can address the namespace and library naming on a new branch.  Those naming changes would extend beyond the files modified in this branch.  @linus-sherrill do you have any objection to that plan?

File: vital/applets/config_validation.h (hunk 1 above)
Comment (linus-sherrill) [reply vital/applets/config_validation.h:41]: These are applet support, so `kwiver::vital::applet` would be appropriate.

**Review by linus-sherrill — APPROVED** [review]
The refactoring looks like a big improvement.

**Pull request merged**